code/modules/mob/living/simple_animal/abnormality/he/wayward_passenger.dm

```diff
@@ -0,0 +1,231 @@
+		return FALSE
+	return ..()
+
+//*** Work mechanics ***
+/mob/living/simple_animal/hostile/abnormality/wayward/FailureEffect(mob/living/carbon/human/user, work_type, pe)
+	if(prob(65))
+		datum_reference.qliphoth_change(-1)
+	return
+
+/mob/living/simple_animal/hostile/abnormality/wayward/NeutralEffect(mob/living/carbon/human/user, work_type, pe)
+	if(prob(10))
+		datum_reference.qliphoth_change(-1)
+	return
+
+/mob/living/simple_animal/hostile/abnormality/wayward/FearEffectText(mob/affected_mob, level = 0)
+	level = num2text(clamp(level, -1, 4))
+	var/list/result_text_list = list(
+		"-1" = list("I've got this.", "How boring.", "Doesn't even phase me."),
+		"0" = list("Just calm down, do what we always do.", "Just don't lose your head and stick to the manual.", "Focus...", "Just call the squire-... wait, what?", "I've seen that logo somewhere..."),
+		"3" = list("Why do I feel so angry?", "Help me...", "I don't want to die!", "Why does this look familiar?"),
+		"4" = list("Is that... from a wing?!", "No... it can't be...", "WHAT IS THAT THING?!")
+		)
+	return pick(result_text_list[level])
+
+//*** Breach mechanics ***
+
+/mob/living/simple_animal/hostile/abnormality/wayward/BreachEffect(mob/living/carbon/human/user)
+	icon_state = "wayward_breach"
+	playsound(src, 'sound/abnormalities/thunderbird/tbird_zombify.ogg', 45, FALSE, 5)//this is the sound effect used for Tomerry in the lovetown reception
+	..()
+
+//*** Teleport code ***//
+/mob/living/simple_animal/hostile/abnormality/wayward/proc/TryTeleport()
+	if(teleport_cooldown > world.time)
+		return FALSE
+	teleport_cooldown = world.time + teleport_cooldown_time//so it doesn't get called twice by life()
+	icon_state = "wayward_tpstart"
+	playsound(src, 'sound/abnormalities/wayward_passenger/attack1.ogg', 100, 1)
+	var/list/teleport_potential = list()
+	for(var/mob/living/L in GLOB.mob_living_list)
+		if(L.stat == DEAD || L.z != z || L.status_flags & GODMODE || faction_check_mob(L))
+			continue
+		if(!ishuman(L))
+			continue
+		if(ishuman(L))
+			var/mob/living/carbon/human/H = L
+			if(H.is_working)
+				continue
+		teleport_potential += get_turf(L)
+	if(!LAZYLEN(teleport_potential))
+		if(!LAZYLEN(GLOB.department_centers))
+			return
+		var/turf/P = pick(GLOB.department_centers)
+		teleport_potential += P
+	can_act = FALSE
+	LoseTarget()
+	var/turf/teleport_target = pick(teleport_potential)
+	SLEEP_CHECK_DEATH(4)
+	playsound(src, 'sound/abnormalities/wayward_passenger/teleport2.ogg', 100, 1)
+	density = FALSE
+	var/obj/effect/portal/abno_warp/P1 = new(get_turf(src))
+	forceMove(teleport_target)
+	var/obj/effect/portal/abno_warp/P2 = new(teleport_target)
+	P1.link_portal(P2)
+	P2.link_portal(P1)
+	icon_state = "wayward_tpend"
+	playsound(src, 'sound/abnormalities/wayward_passenger/teleport2.ogg', 100, 1)
+	SLEEP_CHECK_DEATH(2 SECONDS) //2 seconds to teleport
+	density = TRUE
+	SLEEP_CHECK_DEATH(4)
+	can_act = TRUE
+	icon_state = "wayward_breach"
+	if(isliving(target))
+		dash(target)
```

Comment:
target is never specifically set here after being lost via "LoseTarget" so it may sometimes just _not_ target someone for a bit after teleporting.